flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/connectors/hive/HiveSource.java

```diff
@@ -46,29 +58,98 @@
 
 	private static final long serialVersionUID = 1L;
 
+	private final JobConfWrapper jobConfWrapper;
+	private final List<String> partitionKeys;
+	private final ContinuousPartitionFetcher<Partition, ?> fetcher;
+	private final HiveTableSource.HiveContinuousPartitionFetcherContext<?> fetcherContext;
+	private final ObjectPath tablePath;
+
 	HiveSource(
 			JobConf jobConf,
+			ObjectPath tablePath,
 			CatalogTable catalogTable,
 			List<HiveTablePartition> partitions,
 			@Nullable Long limit,
 			String hiveVersion,
 			boolean useMapRedReader,
-			boolean isStreamingSource,
+			@Nullable ContinuousEnumerationSettings continuousEnumerationSettings,
```

Comment:
It is better to create a builder for `HiveSource` instead of passing all arguments.

Reply:
I added the builder, but the builder constructor still takes quite a few parameters. That's because the BulkFormat needs to be created in the constructor. And since BulkFormat is determined during construction, it makes no sense to allow modifying `tablePath` or `partitions` later through the builder.